ruby-challenges.rb

```diff
@@ -2,29 +2,72 @@
-# HINT: Google 'ruby get rid of nested arrays'
+# HINT: Google 'ruby get rid of nested arrays'  
 
-us_states = { northwest: ['Washington', 'Oregon', 'Idaho'], southwest: ['California', 'Arizona', 'Nevada'], notheast: ['Maine', 'New Hampshire', 'Rhode Island'] }
-# Expected output: ['Arizona', 'California', 'Idaho', 'Maine', 'Nevada', 'New Hampshire', 'Oregon', 'Rhode Island', 'Washington'] 
+# result_array = us_states.values.to_a.flatten.sort    
+# gives me an output of => ["Arizona", "California", "Idaho", "Maine", "Nevada", "New Hampshire", "Oregon", "Rhode Island", "Washington"] 
 
+# def extract_val(hash)
+    # values = hash.values.flatten.sort
+# end
+# us_states = { northwest: ['Washington', 'Oregon', 'Idaho'], southwest: ['California', 'Arizona', 'Nevada'], northeast: ['Maine', 'New Hampshire', 'Rhode Island'] }
+# Expected output: ['Arizona', 'California', 'Idaho', 'Maine', 'Nevada', 'New Hampshire', 'Oregon', 'Rhode Island', 'Washington'] 
+# result = extract_val(us_states)
 
 # --------------------3a) Create a class called Bike that is initialized with a model, wheels, and current_speed. The default number of wheels is 2. The current_speed should start at 0. Create a bike_info method that returns a sentence with all the data from the bike object.
 
 # Expected output example: 'The Trek bike has 2 wheels and is going 0 mph.'
 
+class Bike 
+    attr_accessor :model, :wheels, :current_speed
+    def initialize (model)
+        @model = model
+        @wheels = 2
+        @current_speed = 0
+
+    end
+
+    def bike_info
+        "The #{:model} has #{:wheels} and is going #{:current_speed} mph."
+    end
+    # -------------------3b) Add the ability to pedal faster and brake. The pedal_faster method should increase the speed by a given amount. The brake method should decrease the speed by a given amount. The bike cannot go negative speeds.
+    def pedal_faster(current_speed)
+        # had to find a way to increase the initial speed and maintain the value, didn't knew if I could use 'Integer' as a incremental value, tried this out and it worked! Happy to actually do some code without referring to chatGPT or google.
+        @current_speed =  @current_speed + current_speed
+    end 
 
+    def pedal_brake(break_speed)
+        # in this scenario you want the speed to reduce but without going to zero, so I did a conditional statement to stop the operation when the speed would go to less than zero.
+        @current_speed = @current_speed - break_speed 
+# Didn't figured it out why I am getting ' 10 ,28 , nil , "negative speed doesn't exist" ' as an output instead of getting 23 I am getting nil, without lines of code 57-59 the code gives me the correct expected values except for the -2 one. Since I still have a lot to cover in this assessment I will keep pressing forward.
+        # if @current_speed < 0
+        #     "Negative speed doesn't exist"
```

Comment:
Remember you still want to set current_speed to something. The lowest it can be is 0